icu4c/source/test/intltest/numbertest_simple.cpp

```diff
@@ -0,0 +1,206 @@
+    IcuTestErrorCode status(*this, "testSign");
+
+    SimpleNumberFormatter snf = SimpleNumberFormatter::forLocale("und", status);
+
+    struct TestCase {
+        int64_t input;
+        USimpleNumberSign sign;
+        const char16_t* expected;
+    } cases[] = {
+        { 1, UNUM_SIMPLE_NUMBER_NO_SIGN, u"1" },
+        { 1, UNUM_SIMPLE_NUMBER_PLUS_SIGN, u"+1" },
+        { 1, UNUM_SIMPLE_NUMBER_MINUS_SIGN, u"-1" },
+        { 0, UNUM_SIMPLE_NUMBER_NO_SIGN, u"0" },
+        { 0, UNUM_SIMPLE_NUMBER_PLUS_SIGN, u"+0" },
+        { 0, UNUM_SIMPLE_NUMBER_MINUS_SIGN, u"-0" },
+        { -1, UNUM_SIMPLE_NUMBER_NO_SIGN, u"1" },
+        { -1, UNUM_SIMPLE_NUMBER_PLUS_SIGN, u"+1" },
+        { -1, UNUM_SIMPLE_NUMBER_MINUS_SIGN, u"-1" },
+    };
+    for (auto& cas : cases) {
+        SimpleNumber num = SimpleNumber::forInt64(cas.input, status);
+        num.setSign(cas.sign, status);
+        auto result = snf.format(std::move(num), status);
+        assertEquals("", cas.expected, result.toTempString(status));
+    }
+}
+
+void SimpleNumberFormatterTest::testCopyMove() {
+    IcuTestErrorCode status(*this, "testCopyMove");
+
+    SimpleNumberFormatter snf0 = SimpleNumberFormatter::forLocale("und", status);
+
+    SimpleNumber sn0 = SimpleNumber::forInt64(55, status);
+    SimpleNumber sn1 = std::move(sn0);
+
+    snf0.format(std::move(sn0), status);
+    status.expectErrorAndReset(U_ILLEGAL_ARGUMENT_ERROR, "Use of moved number");
+
+    assertEquals("Move number constructor",
+        u"55",
+        snf0.format(std::move(sn1), status).toTempString(status));
+
+    SimpleNumber sn2;
+    snf0.format(std::move(sn2), status);
+    status.expectErrorAndReset(U_ILLEGAL_ARGUMENT_ERROR, "Default constructed number");
+
+    sn0 = SimpleNumber::forInt64(44, status);
+
+    assertEquals("Move number assignment",
+        u"44",
+        snf0.format(std::move(sn0), status).toTempString(status));
+
+    SimpleNumberFormatter snf1 = std::move(snf0);
+
+    snf0.format(SimpleNumber::forInt64(22, status), status);
+    status.expectErrorAndReset(U_INVALID_STATE_ERROR, "Use of moved formatter");
+
+    assertEquals("Move formatter constructor",
+        u"33",
+        snf1.format(SimpleNumber::forInt64(33, status), status).toTempString(status));
+
+    SimpleNumberFormatter snf2;
+    snf2.format(SimpleNumber::forInt64(22, status), status);
+    status.expectErrorAndReset(U_INVALID_STATE_ERROR, "Default constructed formatter");
+
+    snf0 = std::move(snf1);
+
+    assertEquals("Move formatter assignment",
+        u"22",
+        snf0.format(SimpleNumber::forInt64(22, status), status).toTempString(status));
+
+    snf0 = SimpleNumberFormatter::forLocale("de", status);
+    sn0 = SimpleNumber::forInt64(22, status);
+    sn0 = SimpleNumber::forInt64(11, status);
+
+    assertEquals("Move assignment with nonempty fields",
+        u"11",
+        snf0.format(std::move(sn0), status).toTempString(status));
+}
+
+void SimpleNumberFormatterTest::testCAPI() {
```

Comment:
Remember to also test each C function at least once from cintltst.
We need that to prove that the API works when called from C.

Of course, you can test the Local...Pointer things only from C++ intltest.

Reply:
Can I do this next week? I'd like to get this PR merged to unblock @Manishearth but I'm probably not going to be able to write and test more code until next week